11-payment-encoding.md

```diff
@@ -216,8 +216,9 @@ A reader:
   description.
   - if a valid `n` field is provided:
     - MUST use the `n` field to validate the signature instead of performing signature recovery.
-  - if there is a valid `s` field:
-    - MUST use that as [`payment_secret`](04-onion-routing.md#tlv_payload-payload-format)
+  - MUST require a valid `s` field to be present
+    - MUST fail the payment if no valid `s` field is found
+    - MUST use the `s` field value as [`payment_secret`](04-onion-routing.md#tlv_payload-payload-format)
```

Comment:
For consistency, I think we should use wording that's similar to the other requirements.  Something like:

```suggestion
  - if a valid `s` field is not provided:
    - MUST fail the payment.
  - otherwise:
    - MUST use the `s` field as [`payment_secret`](04-onion-routing.md#tlv_payload-payload-format)
```

Reply:
Question: "MUST fail the payment if no valid `s` field is found". What would be an invalid `s` field? it should fail anyway, no? It seems like it should be "MUST fail the payment if no `s` field is found"?

Reply:
An `s` field that's not exactly 52 bytes would be invalid.  You're suggested wording would mean the reader should *not* fail if such a field is found.

Reply:
Yes, you're right, thanks